### PR TITLE
Add packages and instructions for building gcc

### PR DESCRIPTION
**Pull request by franred**
This packages are required to build and test gcc with make and make
check.
It does include a file to set up the environmet for building gcc
manually and a README with the instructions to follow to do this.

### CODE REVIEW

File: build-gcc-configuration-files/README [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Instructions to build gcc manually (no omnibus package)
+======================================================
+
+You will need gmp, mpfr and mpc to build gcc and then tell gcc where to find it
+building the omnibus package will do this for you (remembers that this will build
+gcc too but we are looking for developing and testing gcc in a separate tree):
+
+  git clone https://github.com/CodethinkLabs/omnibus-codethink-toolchain.git
+  cd omnibus-codethink-toolchain
+  omnibus build codethink-toolchain --override base_dir:./local workers:8
```
Comment (lukaszkolodziejczyk) [inline build-gcc-configuration-files/README:10]: It might be useful to provide which user should I log in to execute those 3 lines.

Comment (franred) [reply build-gcc-configuration-files/README:10]: That probably won't belong here. But in the other README.

**Comment by lukaszkolodziejczyk** [comment]
Building gcc was not successful; the reason for that was the version of linker being too low (`GNU ld version 2.20.51.0.2-5.47.el6_9.1 20100205`).
One solution for that is using Devtoolset's linker which has a higher version (`GNU ld version 2.28-9.el6`). 
To do that, prepend `/opt/rh/devtoolset-7/root/usr/bin/` to PATH. Do it in `build-gcc-configuration-files/gcc-configure-sourceme` file.

File: build-gcc-configuration-files/README [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+After gathering all the required packages we need to tell gcc where these packages are
+and for that we will need to use the gcc-configure-sourceme file:
+
+  git clone https://github.com/CodethinkLabs/gcc.git
+  cd gcc
+  git checkout codethink/$BRANCH_TO_BUILD
+  cd /home/rpm_omnibus/build-gcc
+  cd build-gcc
```
Comment (lukaszkolodziejczyk) [inline build-gcc-configuration-files/README]: Squeeze lines 18-19 into one.

File: build-gcc-configuration-files/gcc-configure-sourceme [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#
+# To configure gcc to be built or run `make check` please
+# run `source gcc-configure-sourceme` before running your
+# `configure` command when building gcc.
+export CFLAGS="-I/opt/codethink-toolchain/include -O2"
+export CPPFLAGS="-I/opt/codethink-toolchain/include -O2"
+export CXXFLAGS="-I/opt/codethink-toolchain/include -O2"
+export LDFLAGS="-Wl,-rpath,/opt/codethink-toolchain/lib -L/lib64 -L/lib -L/opt/codethink-toolchain/lib64 -L/opt/codethink-toolchain/lib"
+export LD_RUN_PATH="/lib64:/lib:/opt/codethink-toolchain/embedded/lib:/opt/codethink-toolchain/lib64:/opt/codethink-toolchain/lib"
+export PATH="/opt/codethink-toolchain/bin:/opt/codethink-toolchain/embedded/bin:/home/rpm_omnibus/rvm/gems/ruby-2.2.0/bin:/home/rpm_omnibus/rvm/gems/ruby-2.2.0@global/bin:/home/rpm_omnibus/rvm/rubies/ruby-2.2.0/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/rpm_omnibus/rvm/bin:/home/rpm_omnibus/bin:/home/rpm_omnibus/gem/ruby/2.2.0/bin"
```
Comment (lukaszkolodziejczyk) [inline build-gcc-configuration-files/gcc-configure-sourceme]: Here the PATH should be prepend with: `/opt/rh/devtoolset-7/root/usr/bin/`

**Comment by franred** [comment]
I've force pushed the changes mentioned by Lukasz

**Pull request merged**